Source/WebCore/platform/ios/PasteboardIOS.mm

```diff
@@ -305,8 +305,10 @@ static void readURLAlongsideAttachmentIfNecessary(PasteboardWebContentReader& re
 
 static bool shouldTreatAsAttachmentByDefault(const String& typeIdentifier)
 {
-    auto type = [UTType typeWithIdentifier:typeIdentifier];
-    return [type conformsToType:UTTypeVCard] || [type conformsToType:UTTypePDF];
+    RetainPtr type = [UTType typeWithIdentifier:typeIdentifier];
+    return [type conformsToType:UTTypeVCard]
+        || [type conformsToType:UTTypePDF]
+        || [[type preferredMIMEType] isEqualToString:@"text/calendar"];
```

Comment:
Would `[type conformsToType:UTTypeCalendarEvent]` not work?

Reply:
Oh, maybe that's what

> 2. The dropped item does not have a preferred attachment representation type specified.

is about

Reply:
Oh, that's actually about Spotlight not explicitly setting `-[NSItemProvider preferredPresentationStyle]` to `UIPreferredPresentationStyleAttachment`.

I'll check to see if `UTTypeCalendarEvent` works…

Reply:
Yep, that works — I'll check that instead, and factor the code out into a for loop like this:

```
static bool shouldTreatAsAttachmentByDefault(const String& typeIdentifier)
{
    RetainPtr type = [UTType typeWithIdentifier:typeIdentifier];
    for (UTType *attachmentType : std::array { UTTypeVCard, UTTypePDF, UTTypeCalendarEvent }) {
        if ([type conformsToType:attachmentType])
            return true;
    }
    return false;
}
```